easypy/caching.py

```diff
@@ -179,12 +180,14 @@ def inner(*args, **kwargs):
                 result, ts = cache.get(key, NOT_CACHED)
 
                 if expiration > 0:
-                    if result is not NOT_FOUND and time.time() - ts > expiration:
+                    if result is not NOT_FOUND and get_ts_func() - ts > expiration:
                         result = NOT_FOUND
                         del cache[key]
 
                 if result is NOT_FOUND:
-                    ts = time.time()
+                    if log_recalculation:
+                        _logger.debug('time cache expired, calculating new value for %s', name)
+                    ts = get_ts_func()
```

Comment:
@idanarye - what do you mean? how is it possible to call time_func after you run the function? you have to run it before in order to determine whether you should run the function at all...